src/app/zap-templates/templates/app/MatterIDL.zapt

```diff
@@ -0,0 +1,80 @@
+// This IDL was generated automatically by ZAP.
+// It is for view/code review purposes only.
+
+{{#chip_shared_structs}}
+{{#if struct_is_fabric_scoped}}
+[fabric_scoped_by=k{{asUpperCamelCase struct_fabric_idx_field}}]
+{{/if}}
+struct {{name}} {
+  {{#zcl_struct_items}}
+  {{type}} k{{asUpperCamelCase label}} = {{fieldIdentifier}};
+  {{/zcl_struct_items}}
+}
+
+{{/chip_shared_structs}}
+{{!TODO: consider #chip_clusters as iteration point as well.
+{{!      Unsure about the differences}}
+{{#all_user_clusters}}
+{{side}} cluster {{asUpperCamelCase name}} = {{code}} {
+  {{#zcl_enums}}
+  enum {{asUpperCamelCase name}} : {{type}} {
+    {{#zcl_enum_items}}
+    k{{asUpperCamelCase label}} = {{value}};
+    {{/zcl_enum_items}}
+  }
+
+  {{/zcl_enums}}
+  {{#chip_cluster_specific_structs}}
+  {{#if struct_is_fabric_scoped}}
+  [fabric_scoped_by=k{{asUpperCamelCase struct_fabric_idx_field}}]
+  {{/if}}
+  struct {{name}} {
+    {{#zcl_struct_items}}
+    {{type}} k{{asUpperCamelCase label}} = {{fieldIdentifier}};
+    {{/zcl_struct_items}}
+  }
+
+  {{/chip_cluster_specific_structs}}
+  {{#zcl_events}}
+  event {{asUpperCamelCase name}}({{priority}}) = {{code}} {
+    {{#zcl_event_fields}}
+    k{{asUpperCamelCase name}} = {{fieldIdentifier}},
+    {{/zcl_event_fields}}
+  }
+
+  {{/zcl_events}}
+  {{#chip_cluster_attributes}}
+  attribute({{#if isWritableAttribute}}readonly{{else}}writable{{/if}}{{#if isReportableAttribute}}, reportable{{/if}}) {{type}} {{asLowerCamelCase name}} = {{code}};
```

Comment:
A few things worth fixing here:

1) Indicating whether the attribute is nullable or not.
2) For list-typed attributes, `{{type}}` is the type of the list entries.  Probably worth making it clearer when we have a list involved.

Reply:
I am playing a bit of a whack-a-more and list is for sure on my todo (together with max lengths for things like bytes  and strings, default values and others). isList should be simple enough, so I will try to do it.

Reply:
Yeah, a bunch of this could be followups.

Note that it might be worthwhile to have a partial that is reused for struct fields, event fields, command fields, attributes, just like we do for the default value thing, so we don't copy/paste some complicated output all those places.  But followup for that refactor might make sense.